### ISSUE TITLE
A defense of 

### ISSUE THREAD

**Issue by lackhand** [bjorn/tiled#2894]
wangid 0 works great in 1-color wangsets with no further augmentation.
It's functional but not as useful in n-color wangsets (there can be a 0 tile, but you're more likely to want several of them, with separate semantic information for each).
For instance, in the grass/dirt/stone tileset, the choice of which element would be shown for wangid0 is essentially arbitrary, but one could make such a selection ("wangid0 is a patch of non-tiling dirt", for instance).

To support embedded "pools" of color, this feels like a special case of variant tiles.
For instance, in a 1-color "grass" tileset, we might have variant tiles of grass-with-flower or grass-with-stone (in addition to grass-with-nothing). This does not feel to me like a different problem than a 2-color "grass/stone" tileset supporting grass-with-flower or grass-with-stone or a 3-color "grass/stone/flower" supporting that same tile combination.
The problem of non-wang variant tile support is orthogonal to wangid 0, is my point. In all 3 cases, there is some well-defined wangid0 tile for the tileset (arguably a non-tiling grass tile?), and the choice of a variant tile embedded within it is not the wangid's problem.

However, since the wangbrush (/wangeraser) attempts to make painting tiles easier, it would be *nice* if they had semantics that supported selecting/rotating between variants. This feels related to the existing `probability` field in some ways; given multiple choices for wangid 0 (or wangid 0x1010...! Or 0x1100...! Any wangid!), the algorithm to select which one should be placed could depend on enough inputs to make this feel natural.
1) wangbrush. When the wangbrush paints the center of a tile with a wangcolor, the variant selected should

**Comment by lackhand** [comment]
Ugh. I meant to abandon this.

**Comment by bjorn** [comment]
I thought it was very interesting, and was a little sad to that your thoughts seemed unfinished. :-)

How do deal with the WangID 0 tiles is still an open question to me, and it would be nice to have an answer this before the next update. There's still a bunch of other things to do as well though. The next development snapshot will bring primarily the re-introduction of explicit corner, edge or mixed Wang sets, which simplifies the setup for the non-mixed cases.

**Comment by lackhand** [comment]
I abandoned it because I ended up talking myself out of it :)

TL;DR: I think wangId0 support requires going to 9 bytes (or supporting 128 colors instead of 256 and using 7 bits per side -> 7*9=63, leaving you one bit for control).

The thought I abandoned here justified viewing the center as "more like" a variant tile than a unique wang tile.
The more I thought about it, though, the worse it seemed, until I decided to abandon post.

To complete the thought, since it seems the algorithm got cut off halfway, I was going to propose (BUT NO LONGER ENDORSE) a tiered probabilistic selection scheme among variants:
When the wangbrush paints a tile center, select among wangid-compatible tiles which share the same color first. If there are none, select among wangid-compatible tiles sharing the same wangset.
When the wangbrush paints a tile edge/corner, follow the above algorithm for each tile's *current* center color; if there are none, then try again using the wangbrush's color.
When the wangbrush paints a 3x3 neighborhood, follow the edge/corner algorithm for each neighbor.

I think I still endorse this algorithm, but I don't endorse treating the center tile as "special" -- it just seems like another part of wangtile matching, afaict, and so seems worth representing that way. Or, I mean, not; I decided not to comment on purpose :-D
